include/alpaka/core/Utility.hpp

```diff
@@ -29,4 +29,41 @@ namespace alpaka::core
 #else
     using std::declval;
 #endif
+
+    /// Returns the ceiling of a / b, as integer.
+    template<typename Integral>
+    [[nodiscard]] ALPAKA_FN_HOST_ACC constexpr auto divCeil(Integral a, Integral b) -> Integral
+    {
+        return (a + b - 1) / b;
+    }
+
+    /// Computes the nth power of base, in integers.
+    template<typename Integral>
+    [[nodiscard]] ALPAKA_FN_HOST_ACC constexpr auto intPow(Integral base, Integral n) -> Integral
+    {
+        if (n == 0)
+            return 1;
+        auto r = base;
+        for(Integral i = 1; i < n; i++)
+            r *= base;
+        return r;
+    }
+
+    /// Computes the floor of the nth root of value, in integers.
+    template<typename Integral>
+    [[nodiscard]] ALPAKA_FN_HOST_ACC constexpr auto nthRootFloor(Integral value, Integral n) -> Integral
+    {
+        // adapted from: https://en.wikipedia.org/wiki/Integer_square_root
+        Integral L = 0;
+        Integral R = value + 1;
+        while(L != R - 1)
+        {
+            Integral const M = (L + R) / 2;
+            if(intPow(M, n) <= value)
+                L = M;
+            else
+                R = M;
+        }
+        return L;
+    }
```

Comment:
In case someone wonders whether that is slower than `std::pow(static_cast<double>(value), 1.0 / static_cast<double>(n))`, it is actually 11x faster: https://quick-bench.com/q/xIDwFmTdzQVETk1jUwkpyvuoEao